app/scripts/controllers/permissions/selectors.js

```diff
@@ -44,6 +45,66 @@ export const getPermittedAccountsByOrigin = createSelector(
   },
 );
 
+/**
+ * Get the permitted accounts for the given scopes by origin
+ *
+ * @param {Record<string, Record<string, unknown>>} state - The PermissionController state
+ * @param {string[]} scopes - The scopes to get the permitted accounts for
+ * @returns {Map<string, string[]>} A map of origins to permitted accounts for the given scopes
+ */
+export const getPermittedAccountsForScopesByOrigin = createSelector(
+  getSubjects,
+  (_, scopes) => scopes,
+  (subjects, scopes) => {
+    return Object.values(subjects).reduce((originToAccountsMap, subject) => {
```

Comment:
`.forEach` could work here in the same way as the selector bellow.

Reply:
Done here: https://github.com/MetaMask/metamask-extension/pull/29887/commits/cebec5d0aa954bc7890086fd0df37f28276fd1c6